### ISSUE TITLE
My class BGSpriteDance that extends BGSprite won't animate

### ISSUE THREAD

**Issue by dombomb64** [ShadowMario/FNF-PsychEngine#11317]
### Describe your problem here.

So, I'm working on converting my mod to source, but I'm stuck on getting an animated sprite to work. It shows up and the `dance` function traces to PowerShell properly, but the actual calls to animate it do nothing. `BGSprite` works fine, though.

Code:

Definition in PlayState:
```
// Outside of any functions
var chaseCouchBg3:BGSpriteDance;

// Meanwhile inside the stage setup
chaseCouchBg3 = new BGSpriteDance('chaseCouchBg3', -1050, -1600, 1, 1, ['Dance Left', 'Dance Right'], true);
```

BGSpriteDance.hx

```
package;

import flixel.FlxSprite;
import flixel.graphics.frames.FlxAtlasFrames;

class BGSpriteDance extends BGSprite
{
	public var leftIdle(default, set):String;
	public var rightIdle(default, set):String;

	public var danced:Bool = false;

	private function set_leftIdle(left:String) {
		leftIdle = left;
		animation.addByPrefix(left, left, 24, false);
		return left;
	}

	private function set_rightIdle(right:String) {
		rightIdle = right;
		animation.addByPrefix(right, right, 24, false);
		return right;
	}

	public function new(image:String, x:Float = 0, y:Float = 0, ?scrollX:Float = 1, ?scrollY:Float = 1, ?animArray:Array<String> = null, ?loop:Bool = false) {
		super(image, x, y, scrollX, scrollY, null, loop);
		//animation.destroyAnimations();
		
		if (animArray != null) {
			frames = Paths.getSparrowAtlas(image);
			for (i in 0...animArray.length) {
				var anim:String = animArray[i];
				animation.addByPrefix(anim, anim, 24, loop);
				if (leftIdle == null) {
					leftIdle = anim;
					animation.play(anim);
					//animation.addByPrefix('danceLeft', anim, 24, loop);
				}
				else if (rightIdle == null) {
					rightIdle = anim;
					//animation.addByPrefix('danceRight', anim, 24, loop);
				}
				//else
					//animation.addByPrefix(anim, anim, 24, loop);
			}
		} else {
			if(image != null) {
				loadGraphic(Paths.image(image));
			}
			active = false;
		}
		scrollFactor.set(scrollX, scrollY);
		antialiasing = ClientPrefs.globalAntialiasing;
		//trace(animation);
	}

	public override function dance(?forceplay:Bool = false) {
		danced = !danced;
		if (danced && leftIdle != null) {
			animation.play(leftIdle, forceplay);
			trace('sliiide to the left');
		}
		else if (!danced && rightIdle != null) {
			animation.play(rightIdle, forceplay);
			trace('sliiide to the right');
		}
		//else if (idleAnim != null) {
			//animation.play(idleAnim, forceplay);
		//}
		trace('could you repeat that');
		//trace(animation.curAnim.name);
	}
}
```

### Are you modding a build from source or with Lua?

Source

### What is your build target?

Windows x64

### Did you edit anything in this build? If so, mention or summarize your changes.

I've edited lots of things like Alphabet.hx and PlayState.hx, but nothing that would mess with the animations.

**Comment by dombomb64** [comment]
So, I've worked on this some more, and here's what I've found.

- Before the `dance()` function is run, the previous animation is neither finished nor paused, but is stuck on frame 0 out of 4.
- The framerate is at a normal 24.
- The left and right animations are not `null`.
- There is nothing in PlayState that is running `finish()`, `pause()`, or `stop()` on the `animation` variable from any `BGSpriteDance`.

It still doesn't animate though.

**Comment by dombomb64** [comment]
I GOT IT TO WORK! So, basically, the `super()` call was being fed `null` for the `animArray` argument because I saw that the code for adding the animations and setting `idleAnim` didn't run if it was `null`, but in BGSprite.hx, the constructor said that if `animArray` wasn't `null`, then it would instead run the `loadGraphic()` function with the `Animated` argument as `false`, which is something that my code did not undo. So, what I had to do was supply the `super()` call with the proper `animArray` value and uncomment the call to `animation.destroyAnimations()`.

TL;DR: It works now! I had to feed the `super()` call the actual correct information and uncomment the line after.

P.S. Actually, I'll probably just make it so that it keeps the animations instead of recreating them.
